src/lib/flow/sol-flow-js.c

```diff
@@ -0,0 +1,1422 @@
+        value.min = duk_require_int(ctx, -3);
+        value.max = duk_require_int(ctx, -2);
+        value.step = duk_require_int(ctx, -1);
+
+        duk_pop_n(ctx, 4); /* step, max, min, val values */
+    }
+
+    sol_flow_send_irange_packet(node, port, &value);
+
+    return 0;
+}
+
+static duk_ret_t
+send_rgb_packet(struct sol_flow_node *node, uint16_t port, duk_context *ctx)
+{
+    struct sol_rgb value;
+
+    duk_require_object_coercible(ctx, 1);
+
+    duk_get_prop_string(ctx, -1, "red");
+    duk_get_prop_string(ctx, -2, "green");
+    duk_get_prop_string(ctx, -3, "blue");
+    duk_get_prop_string(ctx, -4, "red_max");
+    duk_get_prop_string(ctx, -5, "green_max");
+    duk_get_prop_string(ctx, -6, "blue_max");
+
+    value.red = duk_require_int(ctx, -6);
+    value.green = duk_require_int(ctx, -5);
+    value.blue = duk_require_int(ctx, -4);
+    value.red_max = duk_require_int(ctx, -3);
+    value.green_max = duk_require_int(ctx, -2);
+    value.blue_max = duk_require_int(ctx, -1);
+
+    duk_pop_n(ctx, 6); /* blue_max, green_max, red_max, blue, green, red values */
+
+    sol_flow_send_rgb_packet(node, port, &value);
+
+    return 0;
+}
+
+static duk_ret_t
+send_string_packet(struct sol_flow_node *node, uint16_t port, duk_context *ctx)
+{
+    const char *value = duk_require_string(ctx, 1);
+
+    sol_flow_send_string_packet(node, port, value);
+
+    return 0;
+}
+
+static struct sol_flow_node *
+get_node_from_duk_ctx(duk_context *ctx)
+{
+    struct sol_flow_node *n;
+
+    duk_push_global_object(ctx);
+
+    duk_get_prop_string(ctx, -1, "\xFF" "Soletta_node_pointer");
+    n = duk_require_pointer(ctx, -1);
+
+    duk_pop_2(ctx); /* Soletta_node_pointer, global object values */
+
+    return n;
+}
+
+static int
+get_output_port_number(const struct flow_js_type *type, const char *port_name)
+{
+    struct flow_js_port_out *p;
+    uint16_t i;
+
+    SOL_VECTOR_FOREACH_IDX (&type->ports_out, p, i) {
+        if (streq(p->name, port_name))
+            return i;
+    }
+
+    return -EINVAL;
+}
+
+static duk_ret_t
+send_packet(duk_context *ctx)
+{
+    const struct flow_js_port_out *port;
+    const struct flow_js_type *type;
+    const char *port_name;
+    struct sol_flow_node *node;
+    int port_number;
+
+    port_name = duk_require_string(ctx, 0);
+
+    node = get_node_from_duk_ctx(ctx);
+    type = (struct flow_js_type *)node->type;
+    if (!node || !type) {
+        SOL_WRN("Couldn't send packet to '%s' port.", port_name);
+        return 0;
+    }
+
+    /* TODO: Check a cheaper way to do this, if we had hashes we could
+     * use them here. */
+
+    port_number = get_output_port_number(type, port_name);
+    if (port_number < 0) {
+        SOL_ERR("'%s' invalid port name.", port_name);
+        return 0;
+    }
+
+    port = sol_vector_get(&type->ports_out, port_number);
+    if (!port) {
+        SOL_ERR("'%s' invalid port name.", port_name);
+        return 0;
+    }
+
+    if (streq(port->type_name, "boolean"))
```

Comment:
Instead of comparing the port type names, can't we just compare the packet type pointers instead? It's a lot cheaper.